### PR TITLE
libsndfile: Revert to autotools, keep 1.0.31

### PR DESCRIPTION
**Pull request by kode54**
* Official upstream kept at new site
	* Upstream advises to use autotools on macOS
	* Remove legacy patches
	* Use .tar.bz2 from releases archive, pre-bootstrapped
	* Fixes: https://trac.macports.org/ticket/62896

#### Description

Reverts the mess I caused with my switch to CMake. I didn't realize there was a proper way to do the autotools package, including fetching a pre-bootstrapped archive from Github.

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 11.3.1 20E241 arm64
Xcode 12.5 12E262

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: audio/libsndfile/Portfile [hunk 1]
```diff
@@ -1,47 +1,55 @@
...
-      size    662676
-depends_build-append  port:pkgconfig
-depends_lib           port:flac port:libogg port:libvorbis
+    rmd160  ae3fc5bbcb10a034f3edc1240acacd9f1ec349a7 \
+    sha256  a8cfb1c09ea6e90eff4ca87322d4168cdbe5035cb48717b40bf77e751cc02163 \
+    size    875335
+
+github.tarball_from releases
+master_sites        https://github.com/libsndfile/libsndfile/releases/download/${version}
+distfiles           libsndfile-${version}.tar.bz2
```
Comment (reneeotten) [inline audio/libsndfile/Portfile:31]: `distfiles` and `master_sites` should be set automatically by the github PG - did this not work for you (i.e., did you try without these lines)?

File: audio/libsndfile/Portfile [hunk 2]
```diff
@@ -1,47 +1,55 @@
...
+    --enable-external-libs
 
 variant no_external_libs description {Disable FLAC, Ogg and Vorbis support} {
     depends_build-delete  port:pkgconfig
-    depends_lib-delete    port:flac port:libogg port:libvorbis
-    configure.args-append -DENABLE_EXTERNAL_LIBS=OFF
-    configure.args-delete -DENABLE_EXTERNAL_LIBS=ON
+    depends_lib-delete    port:flac port:libogg port:libvorbis port:libopus
+    configure.args-append --disable-external-libs
+    configure.args-delete --enable-external-libs
```
Comment (reneeotten) [inline audio/libsndfile/Portfile:48]: please use `configure.args-replace` instead

File: audio/libsndfile/Portfile [hunk 3]
```diff
@@ -1,47 +1,55 @@
...
 }
 
-variant experimental description {Enable experimental support for Opus, Speex} {
-    depends_lib-append    port:libopus port:speex
-    configure.args-append -DENABLE_EXPERIMENTAL=ON
-    configure.args-delete -DENABLE_EXPERIMENTAL=OFF
+variant octave description {Enable support for Octave} {
+    depends_lib-append    path:bin/octave:octave
+    configure.args-append --enable-octave
+    configure.args-delete --disable-octave
```
Comment (reneeotten) [inline audio/libsndfile/Portfile:54]: same as above

File: audio/libsndfile/Portfile [hunk 4]
```diff
@@ -1,47 +1,55 @@
...
+
 checksums           \
-      rmd160  8663673c3f8eaf463c7b499c732c53c9bba03c04 \
-      sha256  f082718e84a27d99c0e349151344b71dc2032e7aa2d799af0da48a8a270aa06a \
-      size    662676
-depends_build-append  port:pkgconfig
-depends_lib           port:flac port:libogg port:libvorbis
+    rmd160  ae3fc5bbcb10a034f3edc1240acacd9f1ec349a7 \
+    sha256  a8cfb1c09ea6e90eff4ca87322d4168cdbe5035cb48717b40bf77e751cc02163 \
+    size    875335
```
Comment (reneeotten) [inline audio/libsndfile/Portfile:27]: please have the checksums aligned as the rest of the Portfile

File: audio/libsndfile/Portfile (hunk 1 above)
Comment (kode54) [reply audio/libsndfile/Portfile:31]: This was discussed before, the last time I updated this Portfile. The Github PG would just supply it as the Github page, not the actual home page. Or are you saying it would automatically know the link I supplied?

File: audio/libsndfile/Portfile (hunk 2 above)
Comment (kode54) [reply audio/libsndfile/Portfile:48]: This is literally verbatim what the Portfile from 2019 did. Have the standards changed since then? I was not aware there was such a thing as args-replace.

File: audio/libsndfile/Portfile (hunk 1 above)
Comment (reneeotten) [reply audio/libsndfile/Portfile:31]: I am not sure I understand your comment - the fields I mentioned have nothing to do with setting the homepage, or...?

File: audio/libsndfile/Portfile (hunk 4 above)
Comment (kode54) [reply audio/libsndfile/Portfile:27]: If you like, I can re-align the entire file to the same spacing.

Comment (reneeotten) [reply audio/libsndfile/Portfile:27]: that's not what I am asking... The majority of the Portfiles have the checksums aligned in the way that `port -dv checksum <portname>` gives them - there is no real eason to start them on a newline.

File: audio/libsndfile/Portfile [hunk 5]
```diff
@@ -1,47 +1,55 @@
...
     depends_build-delete  port:pkgconfig
-    depends_lib-delete    port:flac port:libogg port:libvorbis
-    configure.args-append -DENABLE_EXTERNAL_LIBS=OFF
-    configure.args-delete -DENABLE_EXTERNAL_LIBS=ON
+    depends_lib-delete    port:flac port:libogg port:libvorbis port:libopus
+    configure.args-append --disable-external-libs
+    configure.args-delete --enable-external-libs
 }
 
-variant experimental description {Enable experimental support for Opus, Speex} {
```
Comment (reneeotten) [inline audio/libsndfile/Portfile:43]: what's the deal with this variant - why did you remove this? Does the new version not have this option anymore and is it included now by default (in that case the dependencies are probably still needed) or is the support just removed completely?

Comment (kode54) [reply audio/libsndfile/Portfile:43]: I pointlessly added it for one release. Do I need to keep it in case someone just added it to their workflow?

It otherwise automatically detects FLAC, Vorbis, Opus, and even Speex, regardless of the alleged "experimental" status of the latter two.

File: audio/libsndfile/Portfile (hunk 1 above)
Comment (kode54) [reply audio/libsndfile/Portfile:31]: Distfiles is not automatically set. It attempts to fetch a tar.gz of the git repo at the tag, which is not bootstrapped.

File: audio/libsndfile/Portfile (hunk 4 above)
Comment (kode54) [reply audio/libsndfile/Portfile:27]: The last dozen or so times I've attempted to use `port checksum` it simply tells me my checksums are wrong, but refuses to tell me what the correct ones are. I've taken to using the openssl tool to calculate them from the correct download manually.

File: audio/libsndfile/Portfile (hunk 1 above)
Comment (reneeotten) [reply audio/libsndfile/Portfile:31]: I see - if you want the `bz2` file instead then you should add `use_bzip2           yes` to the Portfile

File: audio/libsndfile/Portfile (hunk 4 above)
Comment (reneeotten) [reply audio/libsndfile/Portfile:27]: that's possible - you do need to add the `d` or `v` flags or perhaps both I don't remember to have it show the correct ones

File: audio/libsndfile/Portfile (hunk 5 above)
Comment (reneeotten) [reply audio/libsndfile/Portfile:43]: automatically detection is probably not really what we want as then the end result will depend on whether the user has these other ports installed or not (i.e., not a reproducible build). So if you want to enable the support unconditionally for all user you should add the required dependencies, otherwise the port should be build without the support by default and one can enable it using variants.

File: audio/libsndfile/Portfile (hunk 1 above)
Comment (kode54) [reply audio/libsndfile/Portfile:31]: Oops, I wish I'd read the full documentation, then. Just FYI, there are at least a few Portfiles in the tree that use this same stupid hard coding trick instead of `use_bzip2 yes`. May be worth fixing those as well.

File: audio/libsndfile/Portfile (hunk 2 above)
Comment (kode54) [reply audio/libsndfile/Portfile:48]: Fixed, pushing.

File: audio/libsndfile/Portfile (hunk 3 above)
Comment (kode54) [reply audio/libsndfile/Portfile:54]: Same as above, it is fixed in a commit I am about to squash in.

File: audio/libsndfile/Portfile (hunk 4 above)
Comment (kode54) [reply audio/libsndfile/Portfile:27]: Yeah, I've got that, and the checksums will be indented the same as the rest of the arguments in the file. Hopefully, one indent, 4 spaces, is fine for the keys on the rest of the checksums lines.

File: audio/libsndfile/Portfile (hunk 5 above)
Comment (kode54) [reply audio/libsndfile/Portfile:43]: I've re-added the experimental thing, clarifying what it does this time. Indeed, some of the documentation claimed that Opus was gated behind "experimental", but that's no longer the case. The only thing that appears to still be gated behind `--enable-experimental` now is Speex support. It will still detect Speex, but it won't link in support for it without that flag. If you can suggest a clean way to make sure that `no_external_libs` and `experimental` are mutually exclusive, I'm all ears. Considering it should still build if both are specified, except that the `experimental` variant will do absolutely nothing useful other than adding an unused dependency, since the `--disable-external-libs` switch should disable that effectively anyway.

Comment (reneeotten) [reply audio/libsndfile/Portfile:43]: You could probably do something like:
```variant experimental conflicts no_external_libs description {Enable experimental Speex support} {```
and add a conflict the other way around as well

File: audio/libsndfile/Portfile [hunk 6]
```diff
@@ -1,47 +1,57 @@
...
+    --disable-silent-rules \
+    --disable-sqlite \
+    --disable-test-coverage \
+    --enable-external-libs
+
+variant no_external_libs conflicts experimental description {Disable FLAC, Ogg, Vorbis and Opus support} {
+    depends_build-delete   port:pkgconfig
+    depends_lib-delete     port:flac port:libogg port:libvorbis port:libopus
+    configure.args-replace --enable-external-libs --disable-external-libs
+}
```
Comment (reneeotten) [inline audio/libsndfile/Portfile:47]: please make sure to conform to the modeline regarding white space (should be a multiple of 4) throughout the Portfile

Comment (kode54) [reply audio/libsndfile/Portfile:47]: I don't even know what editor to use to conform to a modeline, or how to make an existing editor conform to a modeline. When I was using Vim, it just inserted tab characters for indents. So I switched back to Visual Studio Code.

Comment (kode54) [reply audio/libsndfile/Portfile:47]: This should be resolved now, I hope.
